src/start_location.cpp

```diff
@@ -369,6 +369,10 @@ void start_location::place_player( player &u ) const
         }
     }
 
+    int posx = u.posx();
+    int posy = u.posy();
+    g->update_map( posx, posy ); // The player's submap must be in the center of the map
```

Comment:
Somehow I missed the other `update_map`. And since its argument shouldn't ever be null, so I agree that it should be a reference.